### DISCUSSION TITLE
Serverless Functions gone mad in January with no change to the code, immediate action needed or I'll be blocked

### DISCUSSION THREAD

**Discussion by akoskm** [vercel/community#1465] [Help]
Hi!

I deployed this app in November, and for some reason, it started failing like this at the beginning of January without making any changes to the code.

I want to avoid being blocked from Vercel and want to comply with the Fair Usage Policy, but I can't find a link to logs on the dashboard where I can see what's happening with the serverless functions and why they are failing.

![image](https://user-images.githubusercontent.com/3111627/217161950-6a962085-01bb-4485-9dcd-eb6e39c7175d.png)

I have a very simple website that consists of about 200 static pages. I have two serverless functions, one for receiving email subscriptions, and the other will be used to send out automated emails, but no one is calling that right now.

When I go to the only function logs I can access, I have something like the following:

![image](https://user-images.githubusercontent.com/3111627/217162476-7620b1dd-3c03-470e-8829-1dcb2eba9b97.png)

which is not super useful (no function named `getReader` in my code, I don't know where this is coming from).

In the meantime, Vercel sends me these messages hourly about my usage quota.

![image](https://user-images.githubusercontent.com/3111627/217162698-d00e507f-2eeb-4c5b-92f3-129d748ad182.png)

Can you help me debug this issue? Thanks!

**Comment by amyegan** [comment]
Hi @akoskm! Is your repo public, or are you able to share a minimal reproduction? That's usually the best way for us to see exactly what is happening and identify a solution specific to this problem.

Some other questions:
- Did this problem happen after a recent deployment, or was the site already deployed for a while before this started?
- What frameworks and libraries are used in your app? (e.g. Next, Vue, Svelte, Node, something else)
- If you're using Next.js, are you also using the new beta `app` directory? If so, downgrading to `13.1.1` seems to be the temporary solution until the cause can be identified and fixed.
  - https://github.com/vercel/next.js/issues/45627
- It sounds like something in your code is using [`ReadableStream`](https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream/getReader). While it doesn't seem likely based on your description, but maybe one of these issues is related? 
  - https://github.com/mozilla/pdf.js/issues/14850
  - https://github.com/web3/web3.js/issues/3044

**Reply by akoskm** [reply]
Hi Amy, thanks for your reply.

I just invited you to my repo. It's not super secret I just haven't open-sourced it yet.

> Did this problem happen after a recent deployment, or was the site already deployed for a while before this started?

The site was deployed & working when I started receiving emails from you that I exceeded my Fair Usage limit.

> What frameworks and libraries are used in your app? (e.g. Next, Vue, Svelte, Node, something else)
```
    "airtable": "^0.11.5",
    "dayjs": "^1.11.6",
    "next": "^13.0.4",
    "node-cache": "^5.1.2",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "remark": "^14.0.2",
    "remark-html": "^15.0.1",
    "showdown": "^2.1.0"
```

> If you're using Next.js, are you also using the new beta app directory? If so, downgrading to 13.1.1 seems to be the temporary solution until the cause can be identified and fixed.

I'll give this a shot.

**Reply by akoskm** [reply]
@amyegan I might have found the cause of the problem on this new logging interface.

I added some logging to request caching, and it turns out something's hitting an URL where my app runs:

![image](https://user-images.githubusercontent.com/3111627/222959358-80046c6a-bce5-4740-8076-9c2e75546420.png)

This URL is `https://beginneritjobs-eum3pserr-akoskm.vercel.app/` ~and I don't know where it comes from.~

Looks like it's some kind of deployment URL:
![image](https://user-images.githubusercontent.com/3111627/222959717-b2e70662-265c-421f-b429-4eb5f17ba4f7.png)

Can I turn this off somehow or make it redirect?

Since nothing is cached there, it starts caching. Stuff that might cause a spike in server usage.

Why does it get hit? I didn't share it with anyone. I just discovered it exists.

Here's what happens when my dedicated URL gets hit:

![image](https://user-images.githubusercontent.com/3111627/222959402-134d4b64-45fa-4909-8ee3-91eb3b8b0b92.png)

And here's my domain configuration:

![image](https://user-images.githubusercontent.com/3111627/222959464-cf6ac42a-02a5-408f-9d54-0befb22f551e.png)

**Reply by amyegan** [reply]
That address is the unique [deployment URL](https://vercel.com/docs/concepts/deployments/generated-urls). Are you using the `VERCEL_URL` environment variable anywhere in your code?

**Reply by akoskm** [reply]
No, I'm not using `VERCEL_URL` anywhere in my code.
